google/ggrc-core
# GGRC-4326 External mapping api

# Issue description

We need to allow creation of external relationships via GGRC Relationships API.

# Steps to test the changes

By default all relationships must be not external.
Relationships from external sources should be displayed in UI as usual.
Check that it's not possible to modify regular relationships by external app user.
Check that it's possible to delete relationships from external sources by user.

# Solution description

Add is_external column to Relationships model.
Add a migration to add is_external column to Relationships table.

Add external app user setting.
Add validations to Relationships model base on external app user.
Add find or create external app user to appengine request loader.

Add unittests.
Add integration tests.


# Sanity checklist


# Migration checklist


# PR Review checklist
